react/src/components/UserProfileSettingModal.tsx

```diff
@@ -0,0 +1,287 @@
+import React from "react";
+import { Modal, Input, Form, Select, Divider, message, Switch } from "antd";
+import { useTranslation } from "react-i18next";
+import { useWebComponentInfo } from "./DefaultProviders";
+import { passwordPattern } from "./ResetPasswordRequired";
+import { useSuspendedBackendaiClient } from "../hooks";
+import { useTanMutation } from "../hooks/reactQueryAlias";
+
+const UserProfileSettingModal : React.FC = () => {
+  const { t } = useTranslation();
+
+  const [form] = Form.useForm();
+
+  const [messageApi, contextHolder] = message.useMessage();
+  const _showSuccessMessage = (successMessage: string) => {
+    messageApi.open({
+      type: "success",
+      content: successMessage
+    });
+  };
+  const _showErrorMessage = (errorMessage: string) => {
+    messageApi.open({
+      type: "error",
+      content: errorMessage
+    });
+  };
+
+  const baiClient = useSuspendedBackendaiClient();
+  let email = baiClient.email;
+  let full_name = baiClient.full_name;
+  let loggedAcount = baiClient._config.accessKey;
+  let totpSupported = false;
+  let totpActivated = false;
+  let selectOptions: any[] = [];
+  let keyPairInfo = {
+    keypairs: [{
+      access_key: "",
+      secret_key: ""
+    }]
+  };
+
+  const _showTotpRule = async() => {
+    totpSupported = baiClient.supports('2FA') && await baiClient.isManagerSupportingTOTP();
+    if (totpSupported) {
+      const userInfo = await baiClient.user.get( baiClient.email, ['totp_activated']);
+      totpActivated = userInfo.user.totp_activated;
+    };
+  };
+
+  const _showKeypairInfo = async(email: string) => {
+    keyPairInfo = await baiClient.keypair.list(email, ["access_key", "secret_key"], true);
+    if (keyPairInfo.keypairs) {
+      for (let i=0; i < keyPairInfo.keypairs.length; i++) {
+        selectOptions.push({value: keyPairInfo.keypairs[i].secret_key, label: keyPairInfo.keypairs[i].access_key})
+      };
+      for(let i=0; i<selectOptions.length; i++) {
```

Comment:
```suggestion
      for (let i = 0; i < selectOptions.length; i++) {
```
If you really want to use for loop, please take care of the code convention.

Reply:
I think I should use this loop.😭
I modified code using prettier!